File: algorithms-and-data-structures/algorithms-and-data-structures-2/data-structures/linear-data-structures/list/list.java
	class List{
		private int[] lista;
		private int tamanho,capacidade;
		public List(int n,int c){
			tam=0;
			lista=new int[n];
			capacidade=c;
		}
		void inserirInicio(int n){
			if(tamanho>=capacidade){
				return;	
			}
			for(int i=tamanho;i>0;i--){
				lista[i]=lista[i-1];
			}
			lista[0]=n;
			tamanho++;
		}
		void inserirFim(int n){
			if(tamanho>=capacidade){
				return;
			}
			lista[tamanho]=n;
			tamanho++;
		}
		void inserirPos(int n,int pos){
			if(tamanho>=capacidade ||pos<0 ||pos>tamanho){
				return;
			}
			for(int i=tamanho;i>pos;i--){
				lista[i]=lista[i-1];
			}
			lista[pos]=n;
			tam++;		
		}
		int removerInicio(){
			if(tamanho==0){
				return;
			}
			int resp=array[0];
			tamanho--;
			for(int i=0;i<tamanho;i++){
				lista[i]=lista[i+1];	
			}
			return resp;
		}	
		int removerFim(){
			if(tamanho==0){
				return;
			}
			return lista[--tamanho];	
		}
		int removerPos(int pos){
			if(tamanho==0 || pos<0|| pos>=tamanho){
				return;
			}
			int resp=lista[pos];
			tamanho--;
			for(int i=pos;i<tamanho;i++){
				lista[i]=lista[i+1];
			}
			return resp;
		}
		void print(){
			System.out.print("[ ");
			for(int i=0;i<tamanho;i++){
				System.out.print(lista[i]+" ");
			}
			System.out.print(" ]");
		}	
	}

File: algorithms-and-data-structures/algorithms-and-data-structures-2/data-structures/linear-data-structures/list/list.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int tam, capacidade;
    int *lista;
} list;

void construtor(list *Struct, int capacidade) {
    (*Struct).tam = 0;
    (*Struct).capacidade = capacidade;
    (*Struct).lista = malloc(capacidade * sizeof(int));
}

void inserirInicio(list *Struct, int n) {
    if ((*Struct).tam >= (*Struct).capacidade) {
        return;
    }

    for (int i = (*Struct).tam; i > 0; i--) {
        (*Struct).lista[i] = (*Struct).lista[i - 1];
    }

    (*Struct).lista[0] = n;
    (*Struct).tam++;
}

void inserirPos(list *Struct, int n, int pos) {
    if ((*Struct).tam >= (*Struct).capacidade ||
        pos < 0 || pos > (*Struct).tam) {
        return;
    }

    for (int i = (*Struct).tam; i > pos; i--) {
        (*Struct).lista[i] = (*Struct).lista[i - 1];
    }

    (*Struct).lista[pos] = n;
    (*Struct).tam++;
}

void inserirFim(list *Struct, int n) {
    if ((*Struct).tam >= (*Struct).capacidade) {
        return;
    }

    (*Struct).lista[(*Struct).tam] = n;
    (*Struct).tam++;
}

int removerInicio(list *Struct) {
    if ((*Struct).tam <= 0) {
        return -1;
    }

    int resp = (*Struct).lista[0];

    for (int i = 0; i < (*Struct).tam - 1; i++) {
        (*Struct).lista[i] = (*Struct).lista[i + 1];
    }

    (*Struct).tam--;

    return resp;
}

int removerPos(list *Struct, int pos) {
    if (pos < 0 || pos >= (*Struct).tam) {
        return -1;
    }

    int resp = (*Struct).lista[pos];

    for (int i = pos; i < (*Struct).tam - 1; i++) {
        (*Struct).lista[i] = (*Struct).lista[i + 1];
    }

    (*Struct).tam--;

    return resp;
}

int removerFim(list *Struct) {
    if ((*Struct).tam <= 0) {
        return -1;
    }

    int resp = (*Struct).lista[(*Struct).tam - 1];

    (*Struct).tam--;

    return resp;
}

void print(list *Struct) {
    for (int i = 0; i < (*Struct).tam; i++) {
        printf("%d ", (*Struct).lista[i]);
    }

    printf("\n");
}
